src/components/smart_objects/include/smart_objects/number_schema_item.h

```diff
@@ -157,10 +157,17 @@ errors::eType TNumberSchemaItem<NumberType>::validate(
     report->set_validation_info(validation_info);
     return errors::INVALID_VALUE;
   }
+
   NumberType value(0);
   if (typeid(int32_t) == typeid(value)) {
+    if (Object.asInt() > std::numeric_limits<int32_t>::max() ||
+        Object.asInt() < std::numeric_limits<int32_t>::min())
+      return errors::OUT_OF_RANGE;
     value = utils::SafeStaticCast<int64_t, int32_t>(Object.asInt());
   } else if (typeid(uint32_t) == typeid(value)) {
+    if (Object.asUInt() > std::numeric_limits<uint32_t>::max() ||
+        Object.asUInt() < std::numeric_limits<uint32_t>::min())
+      return errors::OUT_OF_RANGE;
     value = utils::SafeStaticCast<uint64_t, uint32_t>(Object.asUInt());
```

Comment:
@RomanReznichenkoLuxoft do we still need this function? Probably we can simply use `static_cast` here?